platform/jobs/edxEndToEndRepoMasterWeeklyJob.groovy

```diff
@@ -0,0 +1,143 @@
+Map config = [:]
+Binding bindings = getBinding()
+config.putAll(bindings.getVariables())
+PrintStream out = config['out']
+
+
+/* Map to hold the k:v pairs parsed from the secret file */
+Map secretMap = [:]
+try {
+    out.println('Parsing secret YAML file')
+    /* Parse k:v pairs from the secret file referenced by secretFileVariable */
+    String contents = new File("${EDX_END_TO_END_TESTS}").text
+    Yaml yaml = new Yaml()
+    secretMap = yaml.load(contents)
+    out.println('Successfully parsed secret YAML file')
+}
+catch (any) {
+    out.println('Jenkins DSL: Error parsing secret YAML file')
+    out.println('Exiting with error code 1')
+    return 1
+}
+
+/* Iterate over the job configurations */
+secretMap.each { jobConfigs ->
+
+    Map jobConfig = jobConfigs.getValue()
+
+    /* Test secret contains all necessary keys for this job */
+    /* TODO: Use/Build a more robust test framework for this */
+    assert jobConfig.containsKey('authUser')
+    assert jobConfig.containsKey('authPass')
+    assert jobConfig.containsKey('loginEmail')
+    assert jobConfig.containsKey('loginPass')
+    assert jobConfig.containsKey('courseOrg')
+    assert jobConfig.containsKey('courseNumber')
+    assert jobConfig.containsKey('courseRun')
+    assert jobConfig.containsKey('courseDisp')
+    assert jobConfig.containsKey('repoName')
+    assert jobConfig.containsKey('open')
+    assert jobConfig.containsKey('email')
+    assert jobConfig.containsKey('platformUrl')
+    assert jobConfig.containsKey('jobName')
+    assert jobConfig.containsKey('platformCredential')
+
+    job(jobConfig['jobName']) {
+
+        /* For non-open jobs, enable project based security */
+        if (!jobConfig['open'].toBoolean()) {
+            authorization {
+                blocksInheritance(true)
+                permissionAll('edx')
+            }
+        }
+
+        properties {
+              githubProjectUrl(JENKINS_PUBLIC_GITHUB_BASEURL + jobConfig['platformUrl'])
+        }
+
+        logRotator JENKINS_PUBLIC_LOG_ROTATOR() //Discard build after 14 days
+
+        concurrentBuild() //concurrent builds can happen
+
+        label('master') //restrict to flow-worker-bokchoy
+
+        checkoutRetryCount(5)
+
+        environmentVariables {
+            env('BASIC_AUTH_USER', jobConfig['authUser'])
+            env('BASIC_AUTH_PASSWORD', jobConfig['authPass'])
+            env('USER_LOGIN_EMAIL', jobConfig['loginEmail'])
+            env('USER_LOGIN_PASSWORD', jobConfig['loginPass'])
+            env('COURSE_ORG', jobConfig['courseNumber'])
+            env('COURSE_NUMBER', jobConfig['authPass'])
+            env('COURSE_RUN', jobConfig['courseRun'])
+            env('COURSE_DISPLAY_NAME', jobConfig['courseDisp'])
+        }
+
+        scm {
+
+           git { //using git on the branch and url, clone, clean before checkout
+                remote {
+                    url(JENKINS_PUBLIC_GITHUB_BASEURL + jobConfig['platformUrl'] + '.git')
```

Comment:
You want this to be the e2e tests url. The way it's written, it'll pull down edx-platform's repo.